### PR TITLE
Fix unaligned read of short_vec pubkey_size in sigverify

### PR DESCRIPTION
**Pull request by ryoqun**
(This is split from #6236 as [requested by @sakridge ](https://github.com/solana-labs/solana/pull/6236#issuecomment-541804480)!)

#### Problem

`sigverify`'s deserialization is wrong in a corner case; most of time it works as intended.

> In short, this old code incorrectly reads bytes serialized from `MessageHeader` (three of `u8`s) as the length (`ShortU16`) of `short_vec`. (Moderate pun intended... :).
> 
> So, if `MessageHeader.required_num_sigs` are above `0x7f` (according to the comment of `Short16`), `pubkey_size` will be `2` and subsequent offset calculation gets bogus, resulting in **false negative results of sigverify**. This have been working because most of time `MessageHeader.required_num_sigs` is very small compared to the `0x7f`.

From https://github.com/solana-labs/solana/pull/6236/files#r333704395

#### Solution

Shift the offset correctly, taking `size_of` `MessageHeader` into account.

### CODE REVIEW

File: core/src/sigverify.rs [hunk 1]
```diff
@@ -342,6 +347,42 @@ mod tests {
...
 
+    #[test]
+    fn test_large_sigs() {
+        // use any large number to be misinterpreted as 2 bytes when decoded as short_vec
+        let required_num_sigs = 214;
+        let actual_num_sigs = 5;
+
+        let message = Message {
+            header: MessageHeader {
+                num_required_signatures: required_num_sigs as u8,
```
[the remark this answers is not in the document]
Comment (ryoqun) [reply core/src/sigverify.rs]: nice nits! https://github.com/solana-labs/solana/pull/6388/commits/082fd71fc393010a759eb256f615085a2b62c6f9

**Comment by ryoqun** [comment]
> One nit, but lgtm
> :shipit:

yay!

**Comment by mvines** [comment]
@ryoqun thanks!

**Comment by sakridge** [comment]
Yes looks good. Thanks for splitting it out into this commit, it was much easier to understand the change.

**Pull request merged**